archaius2-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java

```diff
@@ -15,299 +15,261 @@
-            runtimeLayer.setProperties(provider.get(config));
-        }
- 
-        // Finally, load any cascaded configuration files for the application
-        compositeConfig = loader.newLoader().withCascadeStrategy(archaiusConfiguration.getCascadeStrategy()).load(archaiusConfiguration.getConfigName());
-        if (compositeConfig != null) {
-            applicationLayer.replaceConfig(APPLICATION_LAYER_NAME, compositeConfig);
-        }
-        
-        // Remote layers most likely need some configuration so we load them after application
-        // configuration has been loaded. 
-        for (ConfigSeeder provider : archaiusConfiguration.getRemoteLayerSeeders()) {
-            remoteLayer.addConfig("remote" + idCounter.incrementAndGet(), provider.get(config));
-        }
-        
-        for (ConfigListener listener : archaiusConfiguration.getConfigListeners()) {
-            config.addListener(listener);
-        }
-        
-        return config;
+    public Decoder getDecoder() {
+        return DefaultDecoder.INSTANCE;
     }
-    
-    /**
-     * Top level 'fast' property factory which is bound to the root configuration 
-     * @param root
-     * @return
-     */
+
     @Provides
     @Singleton
-    PropertyFactory getPropertyFactory(Config root) {
-        return DefaultPropertyFactory.from(root);
+    PropertyFactory getPropertyFactory(Config config) {
+        return DefaultPropertyFactory.from(config);
     }
 
     @Provides
     @Singleton
-    ConfigProxyFactory getProxyFactory(ArchaiusConfiguration config, PropertyFactory factory) {
-        return new ConfigProxyFactory(config.getDecoder(), factory);
+    ConfigProxyFactory getProxyFactory(Decoder decoder, PropertyFactory factory) {
+        return new ConfigProxyFactory(decoder, factory);
     }
     
     @Override
-    public String toString() {
-        return getClass().getSimpleName();
+    public int hashCode() {
+        return getClass().hashCode();
     }
-    
+
+    /**
+     * equals() on Module's is used by Guice to dedup bindings in the event that the 
+     * same module is installed more than once as modules are assumed to not be 
+     * stateful.  ArchaiusModule however is stateful in that much of its configuration
+     * is supplied externally to the Guice injector.  Also, initializing the configuration
+     * subsystem is an application concern and as such ArchaiusModule would only 
+     * be installed once by the application itself.
+     */
     @Override
     public boolean equals(Object obj) {
-        return ArchaiusModule.class.equals(obj.getClass());
-    }
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
 
-    @Override
-    public int hashCode() {
-        return ArchaiusModule.class.hashCode();
+        throw new ProvisionException("Only one ArchaiusModule may be installed");
```

Comment:
I felt that this exception may give more details about the root cause than a duplicate binding error.  However, I also agree that it doesn't make sense to throw an exception from equals().  I'll fix this.